curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/664b82025b371a24781fd40a.md

```diff
@@ -0,0 +1,42 @@
+---
+id: 664b82025b371a24781fd40a
+title: Step 1
+challengeType: 1
+dashedName: step-1
+---
+
+# --description--
+
+JavaScript is the programming language that powers the web. Unlike the HTML and CSS you have learned previously, JavaScript is most commonly used to write logic instead of markup.
+
+In this project, you will learn the basics of Javascript and apply those concepts to building a pyramid
+generator. 
+
+A pyramid generator is a program where you can set the type of character, the count for the pyramid, and the
+direction of the pyramid. The program will then generate a pyramid based on those inputs.
+
+Click on the `"Check your code"` button to proceed to the next step and start the project. 
+
+# --hints--
+
+Test 1
+
+```js
+
+```
+
```

Comment:
```suggestion
```
I think this is better than having a dummy test. It should still pass and is less verbose.

Reply:
It actually passes when the camper first visits the page rather than needing the camper to hit the button. Showing up immediately as passed kinda bugs me.

Reply:
I actually wonder if it would be better to have the camper remove all of the code and test that they've done so?

Reply:
Not sure, but I feel like these should be as easy as possible to get through.

The point is to provide a heads up of what they're building, after all, so we don't need them to do anything.

Reply:
I had pushed an implementation of Naomi's idea. Let me know what you think.

Reply:
I don't think putting artificial barriers in the way of the learners is the way to go.  Deleting the code we've put in the editor doesn't teach them anything, it's just a chore.

If we simply delete the tests, it works great except that it shows a congratulatory message. If we modify the jaw to show

![The text "Submit to continue" in the lower jaw](https://github.com/freeCodeCamp/freeCodeCamp/assets/15801806/3f7d41b0-fab6-4c99-9093-2cb0e846fd1d)

then it would look fine.

Reply:
I see your point about the chore. As for now, I've set it to an empty test with the hint text of "Submit to continue". However I have not yet figured out how to set the message on the lower jaw.